app/api/employee.py

```diff
@@ -15,31 +15,26 @@
 def view_task():
     task_id = request.args.get('task_id', default=None)
     if task_id is not None:
-        try:
-            task = mongo.db.tasks.find_one({'_id': ObjectId(task_id)})
-            task = serialize_doc(task)
-           
-        except Exception:
-            message = "Invalid ObjectId"
-            return jsonify({"success": False, "message": message}), 400
-        
-        if task is None:
-            message = "Invalid ObjectId"
-                 
-            return jsonify({"success": False, "message": message}), 400
-
-        return jsonify({"success": True, "details": task}), 200
+        task = task_id_is_valid(task_id)
+        print("good",task)
+        if task_id_is_valid(task_id):
+            try:
+                task = mongo.db.tasks.find_one({'_id': ObjectId(task_id)})
+                task = serialize_doc(task)
+                return jsonify({"success": True, "details": task}), 200
+            except Exception as err:
+                return jsonify({"success": True, "details": str(err)}), 200
+        return jsonify({"success": False, "message": "Invalid ObjectId"}), 200
 
     decoded_jwt = token_decode()
     user = mongo.db.users.find_one({'_id': ObjectId(decoded_jwt['user_id'])})
-    all_task = list(mongo.db.tasks.find({'email': user['email']})) 
-    
-    if len(all_task) == 0:
+    all_task_list = list(mongo.db.tasks.find({'email': user['email']}))
+    if len(all_task_list) == 0:
```

Comment:
we don't need to compare with 0, python treats 0, and 1 as boolean, and if the condition check that directly without ==.